### PR TITLE
BF: Try to fix things, in case cloning appears to fail.

### PR DESCRIPTION
**Pull request by bpoldrack**
If cloning partially fails, don't break, but run annex init and annex fsck.

This addresses issue #258 .

### CODE REVIEW

**Comment by hanke** [comment]
````
mih@meiner /mnt/smb/fritz/datalad % PATH=~/hacking/datalad/git/bin:$PATH PYTHONPATH=~/hacking/datalad/git datalad install-handle /tmp/testhd test
2015-10-12 11:22:20,619 [ERROR  ] 'git clone -v /tmp/testhd /mnt/smb/fritz/datalad/test' returned with exit code 128
| stderr: 'Cloning into '/mnt/smb/fritz/datalad/test'...
| done.
| error: unable to create symlink some (Permission denied)
| fatal: unable to checkout working tree
| warning: Clone succeeded, but checkout failed.
| You can inspect what was checked out with 'git status'
| and retry the checkout with 'git checkout -f HEAD'
| 
| ' (gitrepo.py:244)
2015-10-12 11:22:20,620 [ERROR  ] always_commit (AttributeError) (main.py:206)
````

Fails on clone.

**Comment by hanke** [comment]
It wants `self.always_commit`, but it is only assigned later on.

**Comment by bpoldrack** [comment]
Yeah - too quickly done ;)

File: datalad/support/annexrepo.py [hunk 1]
```diff
@@ -113,10 +113,20 @@ def __init__(self, path, url=None, runner=None,
...
+            super(AnnexRepo, self).__init__(path, url, runner=runner,
+                                            create=create)
+        except GitCommandError as e:
+            if create and "Clone succeeded, but checkout failed." in str(e):
+                fix_it = True
+            else:
+                raise e
 
         self.always_commit = always_commit
+        if fix_it:
```
Comment (yarikoptic) [inline datalad/support/annexrepo.py]: Worth at least info or even warning log

Comment (bpoldrack) [inline datalad/support/annexrepo.py]: GitCommandError is logged at that point already

2015-10-12 13:14 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

Comment (yarikoptic) [inline datalad/support/annexrepo.py]: Error is lighter bit not that now some fscking is being done

**Pull request merged**